provider/lxd/provider_test.go

```diff
@@ -120,17 +121,120 @@ func (s *providerSuite) TestFinalizeCloud(c *gc.C) {
 	})
 }
 
-func (s *providerSuite) createProvider(ctrl *gomock.Controller) (environs.CloudFinalizer,
+func (s *providerSuite) createProvider(ctrl *gomock.Controller) (environs.EnvironProvider,
 	*testing.MockProviderCredentials,
+	*lxd.MockLXDInterfaceAddress,
 	*lxd.MockProviderLXDServer,
 ) {
 	server := lxd.NewMockProviderLXDServer(ctrl)
 	creds := testing.NewMockProviderCredentials(ctrl)
+	interfaceAddress := lxd.NewMockLXDInterfaceAddress(ctrl)
 
-	provider := lxd.NewProviderWithMocks(creds, utils.GetAddressForInterface, func() (lxd.ProviderLXDServer, error) {
+	provider := lxd.NewProviderWithMocks(creds, interfaceAddress, func() (lxd.ProviderLXDServer, error) {
 		return server, nil
 	})
-	return provider.(environs.CloudFinalizer), creds, server
+	return provider, creds, interfaceAddress, server
+}
+
+func (s *providerSuite) TestFinalizeCloudWithRemoteProvider(c *gc.C) {
+	if !containerLXD.HasSupport() {
+		c.Skip("To be rewritten during LXD code refactoring for cluster support")
```

Comment:
These skip messages should indicate that we are running on a platform without LXD support.

In any case, should skipping be necessary if we are mocking things out and not relying on a running server? They should be safe to remove.

Reply:
that i don't know, was hoping to sort these after our catchup

Reply:
Just remove them. I'll approve.